addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/UrlParamGuesser.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.paramminer;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpSender;
+
+public class UrlParamGuesser {
+
+    private static final Logger logger = LogManager.getLogger(UrlParamGuesser.class);
+
+    private HttpSender httpSender =
+            new HttpSender(Model.getSingleton().getOptionsParam().getConnectionParam(), true, 17);
```

Comment:
Should add a TODO comment here to use the actual core initiator once targeting >= 2.12